polaris-core/src/main/java/org/apache/polaris/core/storage/azure/AzureLocation.java

```diff
@@ -108,15 +108,24 @@ public boolean isChildOf(@NotNull StorageLocation potentialParent) {
       AzureLocation potentialAzureParent = (AzureLocation) potentialParent;
       if (this.container.equals(potentialAzureParent.container)) {
         if (this.storageAccount.equals(potentialAzureParent.storageAccount)) {
-          String slashTerminatedFilePath = ensureTrailingSlash(this.filePath);
-          String slashTerminatedParentFilePath = ensureTrailingSlash(potentialAzureParent.filePath);
-          return slashTerminatedFilePath.startsWith(slashTerminatedParentFilePath);
+          String formattedFilePath = ensureLeadingSlash(ensureTrailingSlash(this.filePath));
+          String formattedParentFilePath = ensureLeadingSlash(ensureTrailingSlash(potentialAzureParent.filePath));
+          return formattedFilePath.startsWith(formattedParentFilePath);
         }
       }
     }
     return false;
   }
 
+  /** If a path doesn't start with `/`, this will add one */
+  private @NotNull String ensureLeadingSlash(@NotNull String location) {
```

Comment:
Since we always call this along with `ensureTrailingSlash` can we just combine them?

Reply:
`ensureTrailingSlash` is also called from the parent class `StorageLocation`, so I wonder if we should just keep them separate

Reply:
Do we not need to `ensureLeadingSlash` in `StorageLocation`?

Reply:
No, the `StorageLocation` class does the isChildOf comparison by comparing the _full_ location path, so it doesn't make sense to change something like `s3://....` to `/s3://....`

Reply:
Ah, I see... that's right. All this manual-path-parsing stuff is so annoying. Can we just move this up to `StorageLocation` then?

Reply:
Sounds good. Done.